widget/view.controller.js

```diff
@@ -99,113 +120,65 @@
-                  $scope.defaultGlobalSettings[res.key] = { 'recordValue': res.jSONValue, 'recordUUID': res.uuid };
-                });
-                iocExtractionConfigService.deleteGBLVariable(gblVarID);
-              }
-              else {
-                var keyName = gblVarToKeyStoreMapping[item].keystore;
-                var payload = _buildPayload(keyName, null, 'findKeyStore');
-                iocExtractionConfigService.getKeyStoreRecord(payload, 'keys').then(function (response) {
-                  if (response && response['hydra:member'] && response['hydra:member'].length > 0) {
-                    $scope.defaultGlobalSettings[response['hydra:member'][0].key] = { 'recordValue': response['hydra:member'][0].jSONValue, 'recordUUID': response['hydra:member'][0].uuid };
-                  }
-                  else {
-                    var keyName = gblVarToKeyStoreMapping[item].keystore;
-                    var keyValue = gblVarToKeyStoreMapping[item].defaultValue.split(',');
-                    var payload = _buildPayload(keyName, keyValue, 'createKeyStore');
-                    iocExtractionConfigService.createOrUpdateKeyStore(payload, 'keys').then(function (res) {
-                      $scope.defaultGlobalSettings[res.key] = { 'recordValue': res.jSONValue, 'recordUUID': res.uuid };
-                    });
-                  }
-                });
-              }
-            });
-          }
-          regexPatternMapping[gblVarToKeyStoreMapping[item].keystore] = { 'index': gblVarToKeyStoreMapping[item].index, "pattern": gblVarToKeyStoreMapping[item].pattern };
-        });
+      var payload = _buildPayload('%sfsp-excludelist%', null, 'findKeyStore');
+      iocExtractionConfigService.getKeyStoreRecord(payload, 'keys').then(function (keystoreDetails) {
+        if (keystoreDetails && keystoreDetails['hydra:member'] && keystoreDetails['hydra:member'].length > 0) {
+          keystoreDetails['hydra:member'].forEach(function (item) {
+
+            // Check if the global variables exists
+            if (item.jSONValue.globalVariable.length > 0) {
+
+              // Move global variable value to keystore and delete the global variable
+              iocExtractionConfigService.getGlobalVariable(item.jSONValue.globalVariable).then(function (gblVariableDetails) {
+                if (gblVariableDetails && gblVariableDetails['hydra:member'] && gblVariableDetails['hydra:member'].length > 0) {
+                  var keyStoreValue = item.jSONValue;
+                  var gblVarID = gblVariableDetails['hydra:member'][0].id;
+                  keyStoreValue['values'] = gblVariableDetails['hydra:member'][0].value.length > 0 ? [...new Set(gblVariableDetails['hydra:member'][0].value.split(','))] : '';
+                  var payload = _buildPayload(item.key, keyStoreValue, 'createKeyStore');
+                  iocExtractionConfigService.createOrUpdateKeyStore(payload, 'keys').then(function (res) {
```

Comment:
what if creation is failed? global variable will be deleted in that case as well. Please validate if all data is present in the KS var then delete it